tests/delayed_rejection_mh_producer_02.cc

```diff
@@ -0,0 +1,101 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The SampleFlow library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of SampleFlow.
+//
+// ---------------------------------------------------------------------
+
+
+// A simple test for the DelayedRejection Metropolis Hastings producer,
+// copied from metropolis_hasting_producer_04.  This test
+// samples integers between 1 and 100, from a probability
+// distribution that has pi(x)=2^(-x). This happens to be an almost
+// normalized probability distribution because
+//
+//    sum_{x=1}^{100} pi(x)
+//    \approx sum_{x=1}^{\infty} pi(x)
+//    =       sum_{x=1}^{\infty} 2^(-x)
+//    =       1
+//
+// and we neglect the difference.
+//
+// For this case, we can compute the mean value to pretty accuracy:
+//
+//   \bar x \approx sum_{x=1}^{\infty} x pi(x)
+//          =       2
+//
+// We use the Consumers::MeanValue to compute an approximation from
+// 100,000 samples of this value. To do this, we need to convert the
+// integer SampleType to double, so that we can compute mean values
+// with floating point arithmetic.
+
+#include <iostream>
+#include <sampleflow/producers/delayed_rejection_mh.h>
+#include <sampleflow/filters/conversion.h>
+#include <sampleflow/consumers/mean_value.h>
+#include <valarray>
+#include <random>
+#include <cmath>
+
+using SampleType = int;
+
+
+// Use the probability distribution mentioned above
+double log_likelihood (const SampleType &x)
+{
+  return -1. * x * std::log(2.);
+}
+
+// Go to the left or right with equal probability. Wrap around if we
+// get below 1 or beyond 100.
+std::pair<SampleType,double> perturb (const SampleType &x, const std::vector<SampleType> y)
```

Comment:
```suggestion
std::pair<SampleType,double> perturb (const SampleType &x, const std::vector<SampleType> &y)
```

Reply:
Since you just ignore the `y` argument altogether, you might as well not name it in the first place:
```suggestion
std::pair<SampleType,double> perturb (const SampleType &x, const std::vector<SampleType> &)
```